### PR TITLE
Feat/rewrite release versions metric

### PR DESCRIPTION
**Pull request by s2-julceb**
The release_versions metric was based on the pull-request event, which was not good as lot of projects create releases without PRs.

The metric is now based in tag creation, this will still not cover all possible release creation ways but should cover way more than the one before.

### CODE REVIEW

**Review by remmycat — COMMENTED** [review]
Nice work! I noticed a few things

File: src/metrics/release_versions/__tests__/__snapshots__/release-versions.test.ts.snap
[the remark this answers is not in the document]
Comment (s2-julceb) [reply src/metrics/release_versions/__tests__/__snapshots__/release-versions.test.ts.snap:1]: not sure how I overlooked to change the last test 🤦

File: src/metrics/release_versions/__tests__/release-versions.test.ts [hunk 1]
```diff
@@ -5,18 +5,9 @@ import {
...
 
-const octokitResponse = {};
-
-jest.mock("./../../../core/octokit.ts", () => ({
-  __esModule: true,
-  default: {
-    request: () => octokitResponse,
-  },
-}));
-
```
[the remark this answers is not in the document]
Comment (remmycat) [reply src/metrics/release_versions/__tests__/release-versions.test.ts:19]: As long as we call handler the tests are always doing octokit requests even if the metric collection doesn't. (as octokit is used to save the output to github)

File: src/metrics/release_versions/metricsConditions.ts [hunk 2]
```diff
@@ -5,14 +5,21 @@ import {
...
   }
 
-  if (dataEvent.payload.action !== "closed") {
+  if (dataEvent.payload.ref_type !== "tag") {
+    return false;
+  }
+
+  if (!dataEvent.payload.ref.match(semverRegex)) {
     return false;
   }
```
[the remark this answers is not in the document]
Comment (s2-julceb) [reply src/metrics/release_versions/metricsConditions.ts:24]: I did not know about the `clean` function, thanks for the hint.

File: src/metrics/release_versions/getVersionByString.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  if (version === null) {
+    // this error should already be caught by the metric condition and type check of the metric itself
+    throw new Error(
+      `${LogErrors.releaseVersionsUnexpectedInvalidVersion}: ${ref}`
+    );
+  }
+  return pick(
+    ["raw", "major", "minor", "patch", "prerelease", "build", "version"],
+    version
+  );
```
Comment (remmycat) [inline src/metrics/release_versions/getVersionByString.ts:17]: I think pick is unfortunately untyped and returns `any`, so if we want to make typescript compare the types we'd need to type this manually `as Pick<SemVer, "raw" | "major" | ...>`.

Maybe we can also add ramda typings from somewhere, but i'd expect them to cause a few issues in places.

**Review by remmycat — COMMENTED** [review]
Thank you for applying the changes! I had two more nitpicks and found one small inconsistency with the version parsing.

File: src/metrics/release_versions/getVersionByString.ts (hunk 3 above)
Comment (s2-julceb) [reply src/metrics/release_versions/getVersionByString.ts:17]: Good call!

**Review by remmycat — APPROVED** [review]
💯

**Pull request merged**